GitCommands/Settings/AppSettings.cs

```diff
@@ -1215,7 +1215,29 @@ public static Font MonospaceFont
         public static Font Font
         {
             get => GetFont("font", SystemFonts.MessageBoxFont);
-            set => SetFont("font", value);
+            set
+            {
+                SetFont("font", value);
+                _fontByStyle.Clear();
```

Comment:
This here clears the cache when we change Font (through Settings), so future calls to GetFontByStyle will do the right thing. Of course, presently we don't have a mechanism to reload UI based on font change, but this is a step in the right direction, I believe.